python/triton/compiler/code_generator.py

```diff
@@ -986,6 +987,8 @@ def visit_For(self, node):
             for_op = self.builder.create_for_op(lb, ub, step, [arg.handle for arg in init_args])
             if num_stages is not None:
                 for_op.set_attr("tt.num_stages", self.builder.get_int32_attr(num_stages))
+            if loop_unroll_factor is not None and loop_unroll_factor > 1:
```

Comment:
Perhaps we do not need the `loop_unroll_factor > 1` check, which can be deferred to the backend.